### PR TITLE
[MGDOBR-959] - Read executor image name from env

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
https://issues.redhat.com/browse/MGDOBR-959

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/BridgeExecutorServiceImpl.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public class BridgeExecutorServiceImpl implements BridgeExecutorService {
     public static final String KAFKA_ERROR_STRATEGY_IGNORE = "ignore";
     public static final String KAFKA_ERROR_STRATEGY_DLQ = "dead-letter-queue";
 
-    @ConfigProperty(name = "event-bridge.executor.image")
```
Comment (r00ta) [inline shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/BridgeExecutorServiceImpl.java:45]: no need to change this line. we already have the default value specified in `application.properties` and you can keep `event-bridge.executor.image` as property name: if there is an env variable called `SMART_EVENTS_EXECUTOR_IMAGE` the smallrye lib will inject that value by magic.

File: shard-operator/src/main/resources/application.properties [hunk 2]
```diff
@@ -7,7 +7,7 @@ quarkus.container-image.name=shard-operator
 quarkus.container-image.tag=latest
 quarkus.container-image.build=false
 
-event-bridge.executor.image=${EVENT_BRIDGE_EXECUTOR_IMAGE:openbridge/executor:latest}
```
Comment (r00ta) [inline shard-operator/src/main/resources/application.properties:10]: same here

File: shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/BridgeExecutorServiceImpl.java (hunk 1 above)
Comment (vaibhavjainwiz) [reply shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/BridgeExecutorServiceImpl.java:45]: @r00ta 
AFAIK dot(`.`) are converted to underscore(`_`) but I am not sure about hyphen(`-`).
Does they converted into underscore as well?

I also think `SMART_EVENTS_EXECUTOR_IMAGE` is making more sense then `EVENT_BRIDGE_EXECUTOR_IMAGE`. wdyt?

Comment (robpblake) [reply shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/BridgeExecutorServiceImpl.java:45]: @vaibhavjainwiz Just to be clear, the environment property currently supported is `EVENT_BRIDGE_EXECUTOR_IMAGE` . I would suggest we keep that one for now rather than renaming a single property with the `SMART_EVENTS_` prefix.

File: shard-operator/pom.xml [hunk 3]
```diff
@@ -195,12 +193,17 @@
         <activeByDefault>true</activeByDefault>
       </activation>
       <properties>
-        <namespace>default</namespace>
-        <quarkus.openshift.namespace>${namespace}</quarkus.openshift.namespace>
         <quarkus.openshift.deploy>false</quarkus.openshift.deploy>
         <quarkus.profile>openshift</quarkus.profile>
         <event-bridge.k8s.platform>openshift</event-bridge.k8s.platform>
       </properties>
+      <dependencies>
```
Comment (robpblake) [inline shard-operator/pom.xml:200]: @vaibhavjainwiz Is this change related?

**Review by robpblake — COMMENTED** [review]
@vaibhavjainwiz It's not clear to me if all the changes here are related to the original JIRA. Can you provide a bit more context on why some of the changes are needed?

cheers,
Rob

File: shard-operator/src/main/kubernetes/minikube.yml [hunk 4]
```diff
@@ -27,33 +37,63 @@ rules:
...
+      - create
+      - delete
+      - patch
+      - update
   - apiGroups:
       - networking.k8s.io
     resources:
       - ingresses
     verbs:
-      - '*'
```
[the remark this answers is not in the document]
Comment (vaibhavjainwiz) [reply shard-operator/src/main/kubernetes/minikube.yml:36]: use `*` as resource or verbs are not allowed as RBAC in managed tenant on boarding. It have to be specific which resource/verb you want to have.

It not just on boarding, it also a recommended practice.

File: shard-operator/src/main/resources/application.properties [hunk 5]
```diff
@@ -7,7 +7,7 @@ quarkus.container-image.name=shard-operator
...
+quarkus.container-image.builder=jib
```
[the remark this answers is not in the document]
Comment (vaibhavjainwiz) [reply shard-operator/src/main/resources/application.properties:10]: yes, we need to add below dependency to generate openshift specific file.
```
<dependency>
          <groupId>io.quarkus</groupId>
          <artifactId>quarkus-openshift</artifactId>
          <optional>true</optional>
        </dependency>
```
On adding, it give me compilation issue to set `quarkus.container-image.builder` property.

File: shard-operator/pom.xml (hunk 3 above)
Comment (vaibhavjainwiz) [reply shard-operator/pom.xml:200]: we need to add below dependency to generate openshift specific file.
 ```
 <dependency>
          <groupId>io.quarkus</groupId>
          <artifactId>quarkus-openshift</artifactId>
          <optional>true</optional>
        </dependency>
```

File: shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/BridgeExecutorServiceImpl.java (hunk 1 above)
Comment (vaibhavjainwiz) [reply shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/BridgeExecutorServiceImpl.java:45]: I am resetting it to `EVENT_BRIDGE_EXECUTOR_IMAGE`

Comment (vaibhavjainwiz) [reply shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/BridgeExecutorServiceImpl.java]: done

File: shard-operator/src/main/resources/application.properties (hunk 2 above)
Comment (vaibhavjainwiz) [reply shard-operator/src/main/resources/application.properties:10]: done

File: shard-operator/src/main/kubernetes/minikube.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (vaibhavjainwiz) [reply shard-operator/src/main/kubernetes/minikube.yml:36]: If you say, I could create a new Jira and make these changes against it.

File: shard-operator/src/main/kubernetes/minikube.yml [hunk 6]
```diff
@@ -73,3 +113,19 @@ roleRef:
...
+  template:
+    metadata:
+      name: shard-operator
+    spec:
+      containers:
+        - name: shard-operator
+          env:
+            - name: EVENT_BRIDGE_EXECUTOR_IMAGE
+              value: openbridge/executor:latest
+---
```
Comment (r00ta) [inline shard-operator/src/main/kubernetes/minikube.yml:131]: Could you remind me why we need this particular config to be specified in this `minikube.yml`? We have a lot more mandatory configs, why only this one is a problem?

Comment (vaibhavjainwiz) [reply shard-operator/src/main/kubernetes/minikube.yml:131]: because we need to set value of `EVENT_BRIDGE_EXECUTOR_IMAGE` env at time of operator bundle creation.

File: shard-operator/src/main/kubernetes/minikube.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (robpblake) [reply shard-operator/src/main/kubernetes/minikube.yml:36]: @vaibhavjainwiz Yes please. Let's keep the changes in this PR focused exactly on setting the image via environment properties 👍 

cheers,
Rob

File: shard-operator/pom.xml (hunk 3 above)
Comment (vaibhavjainwiz) [reply shard-operator/pom.xml:204]: https://quarkus.io/guides/deploying-to-openshift#bootstrapping-the-project

File: shard-operator/src/main/kubernetes/minikube.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (vaibhavjainwiz) [reply shard-operator/src/main/kubernetes/minikube.yml]: created below PRs to handle it
https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/974
https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/973

File: shard-operator/pom.xml (hunk 3 above)
Comment (robpblake) [reply shard-operator/pom.xml:204]: @vaibhavjainwiz Sorry I think I'm missing too much context to fully understand this PR. 

Is there some process related to our CPaaS builds that needs the additional of this dependency to generate something specific to Openshift?

thanks,
Rob

File: shard-operator/src/main/kubernetes/minikube.yml (hunk 6 above)
Comment (r00ta) [reply shard-operator/src/main/kubernetes/minikube.yml:91]: if we need some specific files for CPaaS I think we should generate them and store them somewhere. For example in our `kustomize` projects we generated the CRD definitions with the quarkus plugin and then stored them in the repository, we do not generate them on the fly every time. 

In any case, we have a lot more **mandatory** configs, the yaml you are going to generate is not usable at all (and actually the executor image has to be replaced in the template you generated)

Comment (r00ta) [reply shard-operator/src/main/kubernetes/minikube.yml:91]: for example also for app-sre we keep the specific yaml files here https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/app-interface/manifests/rhose-fleet-manager.yml

Comment (vaibhavjainwiz) [reply shard-operator/src/main/kubernetes/minikube.yml:91]: maintaining them at different places includes extra effort and lots of chance to mismatch of content. I still doesn't see problem to generate them every time and refer them from same place.

Anyway I am closing this PR and duplicate its content into CPaaS onboarding specific repos. I don't want it to be in endless loop.
